src/V3Ast.h

```diff
@@ -1842,6 +1842,10 @@ class AstNode VL_NOT_FINAL {
                       bool doCheck = true);
     static void dumpTreeFileGdb(const AstNode* nodep, const char* filenamep = nullptr);
 
+    void dumpTreeDotOpConnect(std::ostream& os, AstNode* op, const std::string& name) const;
+    void dumpTreeDotNode(std::ostream& str = std::cout) const;
+    void dumpTreeDot(std::ostream& os = std::cout) const;
+    void dumpTreeDotFile(const string& filename, bool append = false, bool doDump = true);
```

Comment:
Do all of these need to be public, aren't some internal to V3Ast?

Reply:
`dumpTreeDotOpConnect` does not use `this` so can be a static function in V3Ast.cpp (instead of a method on `AstNode`). `dumpTreeDotNode` should be inlined into `dumpTreeDot` (after addig the static function to output the `userNp`s).